plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -70,6 +72,22 @@ data DefaultFun
     | Nop1  -- TODO. These are only used for costing calibration and shouldn't be included in the defaults.
     | Nop2
     | Nop3
+    | Fst
+    | Snd
+    | Null
+    | Head
+    | Tail
+    | ConstrB
+    | MapB
+    | ListB
+    | IB
+    | BB
+    | UnConstrB
+    | UnMapB
+    | UnListB
+    | UnIB
+    | UnBB
+    | ChooseData
```

Comment:
I think we need a naming convention for these otherwise it's quite confusing. Can we have `PairFst`/`ListNull`/`DataConstrB` etc.?

Reply:
Or perhaps a controversial suggestion: 
```
data DefaultFun = 
   ByteStringFun ByteStringFun
   | ListFun ListFun
   | ...

data ListFun = Null | Head | Tail
```
Then we could even potentially define the builtin meanings for the bundles of functions separately
```
instance ToBultinMeaning uni ListFun where
    ...

instance ToBuiltinMeaning uni DefaultFun where
    toBuiltinMeaning (ListFun f) = toBuiltinMeaning f
```

Reply:
I think that makes `deriving (Enum, Ix)` choke, but I'll try it out. Also we can have built-in functions that touch multiple built-in types.

Reply:
Yep, it does and I don't want to mess with these instances manually.

Reply:
Fair enough. Maybe something to think about. In the long run it might be nice to try and modularize these a bit, but maybe it's pointless.